src/ProductView/Header/ConfigurationHeader/ConfigurationHeader.js

```diff
@@ -0,0 +1,89 @@
+import React, { Component } from 'react';
+
+class ConfigurationHeader extends Component {
+    constructor(props) {
+        super(props);
+        this.selectedCodingPlugChanged = this.selectedCodingPlugChanged.bind(this);
+        this.selectedSoftwareVersionChanged = this.selectedSoftwareVersionChanged.bind(this);
+        this.toggleShowAll = this.toggleShowAll.bind(this);
+        this.buildCodingPlugAndSoftwareVersionToIdMapping();
+        let initialCodingPlugVersion = Object.keys(this.cpSwToIdMapping)[0];
+        this.state = {selectedCodingPlug: initialCodingPlugVersion,
+                        selectedSoftwareVersion: Object.keys(this.cpSwToIdMapping[initialCodingPlugVersion])[0],
+                        showAll: true};
+    }
+
+    buildCodingPlugAndSoftwareVersionToIdMapping() {
+        this.cpSwToIdMapping = {};
+        for (let configuration of this.props.configurations) {
+            if(!this.cpSwToIdMapping[configuration.codingPlugId]) {
+                this.cpSwToIdMapping[configuration.codingPlugId] = {};
+            }
+            for (let softwareVersion of configuration.codingPlugSoftwareVersions) {
+                this.cpSwToIdMapping[configuration.codingPlugId][softwareVersion] = configuration.id;
+            }
+        }
+    }
+
+    updateProductConfiguration(newCodingPlug, newSoftwareVersion) {
+        this.props.setProductConfiguration(this.cpSwToIdMapping[newCodingPlug][newSoftwareVersion]);
```

Comment:
Please extract 'this.cpSwToIdMapping[newCodingPlug][newSoftwareVersion]' to a variable.